airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -54,26 +54,33 @@ def __init__(self, authenticator: Optional[Union[AuthBase, HttpAuthenticator]] =
     def cache_filename(self) -> str:
         """
         Override if needed. Return the name of cache file
+        Note that if the environment variable REQUEST_CACHE_PATH is not set, the cache will be in-memory only.
         """
         return f"{self.name}.sqlite"
 
     @property
     def use_cache(self) -> bool:
         """
         Override if needed. If True, all records will be cached.
+        Note that if the environment variable REQUEST_CACHE_PATH is not set, the cache will be in-memory only.
         """
         return False
 
     def request_cache(self) -> requests.Session:
-        cache_dir = Path(os.getenv(ENV_REQUEST_CACHE_PATH))
-        return requests_cache.CachedSession(str(cache_dir / self.cache_filename), backend="sqlite")
+        cache_dir = os.getenv(ENV_REQUEST_CACHE_PATH)
+        # Use in-memory cache if cache_dir is not set
+        # This is a non-obvious interface, but it ensures we don't write sql files when running unit tests
+        if cache_dir:
+            return requests_cache.CachedSession(str(Path(cache_dir) / self.cache_filename), backend="sqlite")  # type: ignore # there are no typeshed stubs for requests_cache
+        else:
+            return requests_cache.CachedSession(backend=requests_cache.SQLiteCache(use_memory=True))  # type: ignore # there are no typeshed stubs for requests_cache
```

Comment:
FYI, the same can be achieved with this
```suggestion
        cache_dir = os.getenv(ENV_REQUEST_CACHE_PATH)
        # Use in-memory cache if cache_dir is not set
        # This is a non-obvious interface, but it ensures we don't write sql files when running unit tests
        if cache_dir:
            sqlite_path = str(Path(cache_dir) / self.cache_filename)
        else:
             sqlite_path = "file::memory:?cache=shared"
         return requests_cache.CachedSession(sqlite_path, backend="sqlite")  # type: ignore # there are no typeshed stubs for requests_cache
```